nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -17,34 +17,68 @@
 
 #include <atomic>
 #include <condition_variable>
+#include <memory>
 #include <string>
 #include <thread>
-#include <memory>
 
+#include "lifecycle_msgs/msg/state.hpp"
+#include "nav2_lifecycle/lifecycle_node.hpp"
+#include "nav2_tasks/task_status.hpp"
 #include "rclcpp/rclcpp.hpp"
 #include "std_msgs/msg/empty.hpp"
-#include "nav2_tasks/task_status.hpp"
 
 namespace nav2_tasks
 {
 
 constexpr std::chrono::milliseconds defaultServerTimeout = std::chrono::milliseconds(5000);
 
-template<class CommandMsg, class ResultMsg>
+template<typename CommandMsg, typename ResultMsg>
 const char * getTaskName();
 
-template<class CommandMsg, class ResultMsg>
-class TaskClient
+template<typename CommandMsg, typename ResultMsg>
+class TaskClient : public nav2_lifecycle::LifecycleHelperInterface
 {
 public:
-  explicit TaskClient(rclcpp::Node::SharedPtr & node)
-  : node_(node)
+  explicit TaskClient(nav2_lifecycle::LifecycleNode::SharedPtr node, bool autoinit = false)
+  : node_(node), autoinit_(autoinit)
   {
     resultReceived_ = false;
     statusReceived_ = false;
 
     statusMsg_ = std::make_shared<StatusMsg>();
 
+    // There are some cases where the TaskClient is used from a context that doesn't have
+    // a lifecycle-style inteface (such as BehaviorTrees). For those situations, the
+    // TaskClient class can be automatically configured and activated
+
+    if (autoinit_) {
+      rclcpp_lifecycle::State state0(
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, "unconfigured");
```

Comment:
Shouldn't this be "configured"?

Reply:
Not sure I understand this either. The states get passed into the on_configure and on_active() methods but the state is not actually used there. I assume that's a redundancy in the lifecycle API? Also I think this should be: 

```
lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED, "unconfigured");
```
I'm assuming you are setting the state to unconfigured and passing into the "onConfigure", similarly below you set it to inactive before activating?

Reply:
In the nav2_robot PR , you did this instead of passing in unused params:
Robot::on_activate(const rclcpp_lifecycle::State & /*state*/)

Can we do that here too or were you expecting to add usage of the state later? Or is there something I'm completely missing?

Reply:
Fortunately, the task_client and task_server are now obsolete and have been removed. No longer an issue with the action client since it doesn't have to be activated (at least so far, as it is not lifecycle aware yet).